wled00/FX.cpp

```diff
@@ -2946,18 +2895,22 @@ uint16_t WS2812FX::mode_starburst(void) {
   }
   return FRAMETIME;
 }
-
+#undef STARBURST_MAX_FRAG
+#undef STARBURST_MAX_STARS
 
 /*
  * Exploding fireworks effect
  * adapted from: http://www.anirama.com/1000leds/1d-fireworks/
  */
-
+#ifdef ESP8266
+  #define MAX_SPARKS 20 // number of fragments
```

Comment:
I think this is a bit few and will makes the effect look far less brilliant in the (likely) case few segments are used.
I will think of a "smart" logic here to enable both running on many segments, and maintaining `min(seglen/2, 80)` sparks for few segments.

Reply:
Many segments and effects with great memory requirements don't go well together. 😄

Reply:
My base use case is a megatree with 16 segments, 50 LEDs each (and a star at the top).
Having a single segment with the same number of LEDs would definitely make it look boring in such case.
But then, when you cross the 600 LEDs barrier you are already in the ESP32 territory where the number of fragments increases.
For the 8266 my actual use case is a megatree with 12 segments, 33 LEDs each (and the star) totaling slightly above 420 LEDs and my roof trim with 4 segments and approximately 600 LEDs. In both cases the effect still looks ok.

If we are to employ dynamic max fragment calculation we should only do it at the time when number of segments change.